wyvern/wyvern_typing.py

```diff
@@ -3,12 +3,13 @@
 
 from pydantic import BaseModel
 
-from wyvern.entities.identifier_entities import WyvernEntity
+from wyvern.entities.identifier_entities import ProductEntity, WyvernEntity
 from wyvern.entities.request import BaseWyvernRequest
 
 T = TypeVar("T")
 REQUEST_ENTITY = TypeVar("REQUEST_ENTITY", bound=BaseWyvernRequest)
 WYVERN_ENTITY = TypeVar("WYVERN_ENTITY", bound=WyvernEntity)
+PRODUCT_ENTITY = TypeVar("PRODUCT_ENTITY", bound=ProductEntity)
```

Comment:
Just curious, why is this called `PRODUCT_ENTITY`? We should really be agnostic to the actual entity being ranked (For example, brand, listing, category, user, carousel, etc)

Reply:
I tried doing it but it was very challenging. i was having trouble defining a generic solution.

I'll try it again now.

Reply:
We can do it later, but we should come back and fix this eventually

Reply:
I'll do it now.

Reply:
Done.  it actually becomes really easy this time lol. it was not easy doing it from the beginning